python/ray/_raylet.pyx

```diff
@@ -3026,13 +3027,30 @@ cdef class CoreWorker:
 
         return c_object_id.Binary()
 
-    def wait(self, object_refs, int num_returns, int64_t timeout_ms,
+    def wait(self, object_refs_or_generator, int num_returns, int64_t timeout_ms,
              TaskID current_task_id, c_bool fetch_local):
         cdef:
             c_vector[CObjectID] wait_ids
             c_vector[c_bool] results
             CTaskID c_task_id = current_task_id.native()
 
+        object_refs = []
```

Comment:
nit: consider pre-allocating array size to avoid high amortized O(1) append? we may see perf regression in ray.wait release test, it waits 10k object refs in a loop IIRC

```python
object_refs = [None] * len(object_refs_or_generator)
for i, ref_or_generator in enumerate(object_refs_or_generator):
  ...
  object_refs[i] = ...

```

fine to ignore, don't want to prematurely optimize